Applications/Spire/Source/Ui/OverlayPanel.cpp

```diff
@@ -217,8 +217,10 @@ void OverlayPanel::on_parent_move(QPoint) {
 }
 
 void OverlayPanel::update_mask() {
-  auto intersection = geometry().intersected(
-    QRect(parentWidget()->mapToGlobal(QPoint()), parentWidget()->size()));
-  setMask(QPolygon(rect()).subtracted(
-    QRect(mapFromGlobal(intersection.topLeft()), intersection.size())));
+  if(m_positioning == Positioning::PARENT) {
```

Comment:
Briefly explain what this fixes.

Reply:
The purpose of the function `update_mask` is to mask the overlapping region between the `OverlayPanel` and its parent in order to reveal the border style of parent component when `Positioning` is `Positioning::PARENT`. To be precise, the area is the shared borders between the `OverlayPanel` and its parent. However, if `Positioning` is `Positioning::NONE`, the `OverlayPanel` might be completely covered by its parent when it is placed inside the area of the parent. Hence, this functionality of `update_mask` is revised to limit to `Positioning = PARENT`.

Reply:
Approved.